solidity/contracts/isms/ccip-read/SP1LightClientIsm.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ External Imports ============
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ISP1LightClient} from "../../interfaces/ISP1LightClient.sol";
+
+// ============ Internal Imports ============
+
+import {AbstractCcipReadIsm} from "./AbstractCcipReadIsm.sol";
+import {Message} from "../../libs/Message.sol";
+import {Mailbox} from "../../Mailbox.sol";
+import {DispatchedHook} from "../../hooks/DispatchedHook.sol";
+import {StorageProof} from "../../libs/StateProofHelpers.sol";
+import {ISuccinctProofsService} from "../../interfaces/ccip-gateways/ISuccinctProofsService.sol";
+
+/**
+ * @title SP1LightClientIsm
+ * @notice Uses Succinct to verify that a message was delivered via a Hyperlane Mailbox and tracked by DispatchedHook
+ */
+contract SP1LightClientIsm is AbstractCcipReadIsm, OwnableUpgradeable {
```

Comment:
I'm a bit out my comfort zone, but IIUC there should be some pretty general logic in here that could be extracted to a "StorageProofIsm"? Specifically, the storage proof logic seems independent of SP1 right?

Reply:
Move generalized logic to StorageProofIsm